scripts/pack-openclaw-tgz.sh

```
@@ -0,0 +1,177 @@
+
+  if [[ -n "$expected" && -f "$SOURCE_DIR/$expected" ]]; then
+    echo "$SOURCE_DIR/$expected"
+    return 0
+  fi
+
+  local candidates=("$SOURCE_DIR"/*.tgz)
+  if [[ ${#candidates[@]} -eq 0 || ! -f "${candidates[0]}" ]]; then
+    return 1
+  fi
+
+  ls -t "$SOURCE_DIR"/*.tgz | sed -n '1p'
```

Shellcheck (0.11.0)

[info] Use find instead of ls to better handle non-alphanumeric filenames.

(SC2012)